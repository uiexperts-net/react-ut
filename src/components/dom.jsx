import React, { useState } from "react"

export const DOMTesting = () => {
    const [isConnected, setConnect] = useState('NO');

    return (
        <div data-testid='root'>
            <h1> DOM Testing</h1>
            <div>Connect Status : <span data-testid='connect-status'>Status:{isConnected}</span></div>
            <button onClick={()=>setConnect('YES')}>Connect</button>
        </div>
    )

}