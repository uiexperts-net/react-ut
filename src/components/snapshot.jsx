import React from 'react';

export const SnapshotComp = ()=>{
    return(
        <div>
            <h1>Hello from Snapshot</h1>
        </div>
    )
}