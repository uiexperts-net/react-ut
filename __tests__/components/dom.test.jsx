import React from 'react';
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import { DOMTesting } from '../../src/components/dom';

describe('unit testing for demo', () => {
  
    test('loads and displays greeting', async () => {
        // ARRANGE
        render(<DOMTesting />)

        // ACT
        await screen.getByTestId('connect-status');

        // ASSERT
        expect(screen.getByTestId('connect-status')).toHaveTextContent('Status:NO');
        await userEvent.click(screen.getByRole('button'))
        expect(screen.getByTestId('connect-status')).toHaveTextContent('Status:YES');
        await userEvent.click(screen.getByRole('button'))
        expect(screen.getByTestId('root')).toMatchSnapshot();      
    })
})