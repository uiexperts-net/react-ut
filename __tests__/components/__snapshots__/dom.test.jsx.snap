// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit testing for demo loads and displays greeting 1`] = `
<div
  data-testid="root"
>
  <h1>
     DOM Testing
  </h1>
  <div>
    Connect Status : 
    <span
      data-testid="connect-status"
    >
      Status:
      YES
    </span>
  </div>
  <button>
    Connect
  </button>
</div>
`;
